viite-backend/viite-main/src/main/scala/fi/liikennevirasto/viite/process/strategy/TerminatedTwoTrackSectionSynchronizer.scala

```diff
@@ -0,0 +1,473 @@
+   * (Homogeneous by RoadPart, Track, Status and Continuous by address M values)
+   */
+  def terminatedLinksToToContinuousTwoTrackSections(terminatedLinks: Seq[ProjectLink]) : (Seq[Seq[ProjectLink]], Seq[Seq[ProjectLink]]) = {
+    val terminatedLeft = terminatedLinks.filter(_.track == Track.LeftSide)
+    val terminatedRight = terminatedLinks.filter(_.track == Track.RightSide)
+    // Group terminated links into continuous sections by status
+    val leftTerminatedSections = toContinuousSectionsByStatus(terminatedLeft, RoadAddressChangeType.Termination)
+    val rightTerminatedSections = toContinuousSectionsByStatus(terminatedRight, RoadAddressChangeType.Termination)
+    (leftTerminatedSections, rightTerminatedSections)
+  }
+
+  /**
+   * Checks if there is a two track termination on road part start.
+   * If there is, then adjusts the terminated tracks to match + the originalAddrMRange start of the links right after the terminated segment.
+   * Else returns the project links unchanged.
+   * @param projectLinks Sequence of project links to adjust
+   */
+  def roadPartStartTerminated(projectLinks: Seq[ProjectLink]): Seq[ProjectLink] = {
+    val terminatedLinks = projectLinks.filter(_.status == RoadAddressChangeType.Termination)
+
+    val (leftTerminatedSections, rightTerminatedSections) = terminatedLinksToToContinuousTwoTrackSections(terminatedLinks)
+
+    val roadPartStartTerminatedLefts = leftTerminatedSections.find(section => section.exists(_.addrMRange.isRoadPartStart))
+    val roadPartStartTerminatedRights = rightTerminatedSections.find(section => section.exists(_.addrMRange.isRoadPartStart))
+
+    val processedLinks = if (roadPartStartTerminatedLefts.nonEmpty && roadPartStartTerminatedRights.nonEmpty) { // Check if road part start is terminated on both tracks
+      handleTwoTrackRoadPartStartTermination(roadPartStartTerminatedLefts.get, roadPartStartTerminatedRights.get, projectLinks)
+    } else
+      projectLinks
+
+    processedLinks
+  }
+
+  /**
+   * Adjusts terminated sections on two track road part start to match (if reasonably close to each other) and
+   *  the links that come after the terminated section.
+   *
+   * Example:
+   *
+   *  Terminated:     ==>
+   *  Nonterminated:  -->
+   *
+   *  Before:
+   *
+   *  0       202  250   450
+   *  ========>---->----->
+   *  ======>------->---->
+   *  0     200     260  450
+   *
+   *  After:
+   *
+   *  0       201  250   450
+   *  ========>---->----->
+   *  ======>------->---->
+   *  0     201     260  450
+   */
+  def handleTwoTrackRoadPartStartTermination(terminatedLeftSection: Seq[ProjectLink], terminatedRightSection: Seq[ProjectLink], projectLinks: Seq[ProjectLink]): Seq[ProjectLink]= {
+    val maxDiffForTrackEnds = 10 // This number is arbitrary and may require adjustments in the future.
+    val lastTerminatedOnLeftSideSection = terminatedLeftSection.last
+    val lastTerminatedOnRightSideSection = terminatedRightSection.last
+
```

Comment:
Toivon tähän kommentin, miksi verrataan alkuperäisiä etäisyyslukemia

Reply:
// We need to modify (==average) the original addresses, as the terminated sections must have equal end address (and they do should not have new address values).